cdc/log.cc

```diff
@@ -266,8 +267,8 @@ class cdc::cdc_service::impl : service::migration_listener::empty_listener {
     }
 };
 
-cdc::cdc_service::cdc_service(service::storage_proxy& proxy)
-    : cdc_service(db_context::builder(proxy).build())
+cdc::cdc_service::cdc_service(service::storage_proxy& proxy, cdc::generation_service& gen_svc)
```

Comment:
I think it would be better not to couple cdc_service with generation_service. You can just take cdc::metadata& and use generation_service::get_cdc_metadata() by the caller of cdc_service constructor.

Reply:
Yeah.
Unfortunately, I'm unsure how to do that elegantly.

It would probably be best if `service` was a [functor](https://en.wikipedia.org/wiki/Functor_(functional_programming)) and we had a function
```
template <typename T1, typename T2>
service<T2> fmap(function<T1, T2> f, service<T1>)
```
which we could use for `f = generation_service::get_cdc_metadata` to obtain `service<cdc::metadata>`, which wouldn't be a "real" service but could be passed to `cdc.start(...)`.

Reply:
I was also thinking of using `sharded<T>::start_single`, but it doesn't allow to start multiple instances one by one:
```
sharded<Service>::start_single(Args&&... args) {
    assert(_instances.empty());
    _instances.resize(1);
```